rust_icu_unum/src/lib.rs

```diff
@@ -443,6 +675,42 @@ impl<'a, T> Iterator for UFieldPositionIterator<'a, T> {
     }
 }
 
+/// Gets an iterator over all available formatting locales.
+///
+/// Implements `unum_getAvailable`. Since 0.3.1.
+pub fn available_iter() -> UnumIter {
+    let max = unsafe { versioned_function!(unum_countAvailable)() } as usize;
+    UnumIter { max, next: 0 }
+}
+
+/// An iterator returned by `available_iter()`, containing the string representation of locales for
+/// which formatting is available.
+pub struct UnumIter {
+    /// The total number of elements that this iterator can yield.
+    max: usize,
+    /// The next element index that can be yielded.
+    next: usize,
+}
+
+impl Iterator for UnumIter {
+    type Item = &'static str;
+
+    fn next(&mut self) -> Option<&'static str> {
+        if self.max == 0 {
+            return None;
+        }
+        if self.next >= self.max {
+            return None;
+        }
+        let c = unsafe {
+            let cptr = versioned_function!(unum_getAvailable)(self.next as i32);
+            std::ffi::CStr::from_ptr(cptr)
```

Comment:
If `cptr` is null, this will segfault.

Reply:
Added an assertion and a comment. I am not sure how we can guard against this possibility.